src/service/payout-request-pdf-service.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ *  SudoSOS back-end API service.
+ *  Copyright (C) 2024  Study association GEWIS
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published
+ *  by the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import { Client, FileResponse } from 'pdf-generator-client';
+import PayoutRequest from '../entity/transactions/payout-request';
+import FileService from './file-service';
+import {
+  PayoutParameters,
+  PayoutRouteParams,
+  Payout,
+  FileSettings,
+  ReturnFileType,
+  Language,
+  IPayoutRouteParams,
+} from 'pdf-generator-client';
+import { PAYOUT_REQUEST_PDF_LOCATION } from '../files/storage';
+import {PdfGenerator} from "../entity/file/pdf-file";
+import PayoutRequestPdf from "../entity/file/payout-request-pdf";
+
+const PDF_GEN_URL =  process.env.PDF_GEN_URL ? process.env.PDF_GEN_URL : 'http://localhost:3001/pdf';
+
+export default class PayoutRequestPdfService {
+
+  static pdfGenerator: PdfGenerator = {
+    client: new Client(PDF_GEN_URL, { fetch }),
+    fileService: new FileService(PAYOUT_REQUEST_PDF_LOCATION),
+  };
+
+
+  /**
+   * Constructs and returns the parameters required for generating a payout request PDF.
+   * @param payoutRequest - The payout request for which to generate the parameters.
+   * @returns {PayoutParameters} An instance of `PayoutParameters` containing all necessary information for PDF generation.
+   */
+  static getParameters(payoutRequest: PayoutRequest): PayoutParameters {
+    return new PayoutParameters({
+      payout: new Payout({
+        bankAccountName: payoutRequest.bankAccountName,
+        bankAccountNumber: payoutRequest.bankAccountNumber,
+        amount: payoutRequest.amount.getAmount(),
+        reference: `SDS-PR-${payoutRequest.id}`,
```

Comment:
Maybe prefix the ID with some zeroes, so the number always has a length of 3 or 4 digits. `SDS-PR-001` looks nicer than `SDS-PR-1`.